Implements GetSpellMinLevel() to fix issue #6063

DotesMiqotes opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->
Fixes issue #6063 by doing the following:
Adds `uint16 GetSpellMinLevel(SpellID spellId)` to `mob_spell_list` that returns the `min_level` of a spell from a mob's spell list.
Changes the trust level check in spell.cpp to use `GetSpellMinLevel()`.

## Steps to test these changes

<!-- Clear and detailed steps to test your changes here -->
Recompiled the server and tested multiple trusts at different levels to make sure they were casting the right spells based on their `min_level` in `mob_spell_lists.sql`.

DotesMiqotes committed:
Implements GetSpellMinLevel() to fix issue #6063

Added `uint16 GetSpellMinLevel(CMobEntity* PMob, SpellID spellId)` to `mobutils` that returns the `min_level` of a spell from a mob's spell list.
Changed the trust level check in spell.cpp to use `GetSpellMinLevel()`.

zach2good commented on src/map/utils/mobutils.cpp:
```diff
@@ -285,6 +285,23 @@ namespace mobutils
+     *  Returns the min_level of a spell from a mob's spell list             *
+     *                                                                       *
+     ************************************************************************/
+    uint16 GetSpellMinLevel(CMobEntity* PMob, SpellID spellId)
```
https://godbolt.org/z/q74cW6c8Y

DotesMiqotes replied:
Thank you, I was missing the ` const` in `auto const&` that was keeping the Ranged-based for loop from working. I was also able to take it a step further based on the code in your godbolt.org link, and moved the function to `mob_spell_list`.

It cleans up everything nicely. I'll also update the details in the PR to reflect the move.

DotesMiqotes committed:
GetSpellMinLevel() Loop Change and Move

Changed the for loop in GetSpellMinLevel() to a ranged-based for loop.
Moved GetSpellMinLevel() from `mobutils` to `mob_spell_list`.

DotesMiqotes committed:
Added 2 spaces to mob_spell_list.h

Added 2 spaces to an unedited line of mob_spell_list.h so it would pass Sanity_Checks.
